nilearn/interfaces/bids/glm.py

```diff
@@ -342,7 +354,11 @@ def save_glm_to_bids(
             )
             contrast_plot.set_xlabel(contrast_name)
             contrast_plot.figure.set_figheight(2)
-            contrast_plot.figure.set_tight_layout(True)
+            # TODO remove if block when bumlping matplotlib to >= 3.6
+            if compare_version(mpl_version, ">=", "3.6.0"):
+                contrast_plot.figure.set_layout_engine("tight")
+            else:
+                contrast_plot.figure.set_tight_layout(True)
```

Comment:
hmmm... this may be an overkill.

I think a simple `tight_layout()` may be enough:

>Note that [matplotlib.pyplot.tight_layout()](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.tight_layout.html#matplotlib.pyplot.tight_layout) will only adjust the subplot params when it is called. In order to perform this adjustment each time the figure is redrawn, you can call fig.set_tight_layout(True), or, equivalently, set [rcParams["figure.autolayout"]](https://matplotlib.org/stable/users/explain/customizing.html?highlight=figure.autolayout#matplotlibrc-sample) (default: False) to True.
> [Source](https://matplotlib.org/stable/users/explain/axes/tight_layout_guide.html#tight-layout-guide)